app/controllers/application_controller.rb

```diff
@@ -331,6 +331,20 @@ def make_dlist(cuds)
     emails.join(",")
   end
 
+  # Retrieves necessary data for autocompletion purposes
+  # Currently used by manage extensions and create submission
+  def retrieve_autocompletion_data!
+    @users = {}
+    @usersEncoded = {}
+    @course.course_user_data.each do |cud|
+      # Escape once here, and another time in _autocomplete.html.erb (see comments)
+      @users[CGI.escapeHTML cud.full_name_with_email] = cud.id
```

Comment:
We escape once here, and once on the frontend because the alternatives are
- Escape twice on backend, but this leads to a yucky `CGI.escapeHTML CGI.escapeHTML ...` on the backend and a `<%= j k.html_safe %>` on the frontend.
- Escape twice on frontend, but there isn't an easy way to do so since `escape_html` sets `html_safe` to true, and so a second `escape_html` would not work to escape the text again.

I think this is the best way to go about it.